third_party/grpc-rust/grpc-compiler/src/codegen.rs

```diff
@@ -153,20 +153,24 @@ impl<'a> MethodGen<'a> {
             match self.proto.get_client_streaming() {
                 false => {
                     param_in = "req";
-                    w.write_line(
-                        "let req = protobuf::parse_from_reader(&mut grpc_pair.receive).unwrap();",
-                    )
+                    w.write_line("let mut buf = Vec::<u8>::with_capacity(256);");
```

Comment:
IIUC this will try to read into a buffer of 256 bytes; if the data fits, it will be read there immediately; if not, the vector will be reallocated to the correct size and then retried. If that's correct, could you add a comment here? Also we should probably enforce a maximum size for both read and write somewhere (maybe just leave a TODO about that somewhere for now).